tests/abi/use_this.cpp

```diff
@@ -0,0 +1,28 @@
+#include "doctest_compatibility.h"
+
+#include <nlohmann/json.hpp>
+
+TEST_CASE("use this")
+{
+    SECTION("implicit")
+    {
+        using nlohmann::json;
+        using nlohmann::ordered_json;
+
+        std::string p = "/root";
+        ordered_json j;
+        j[json::json_pointer(p)] = json::object();
+        CHECK(j.dump() == "{\"root\":{}}");
```

Comment:
These tests rely on `json_pointer` implicitly converting to string in v3.10.5. I should add comments.